src/cmap/auth/mongodb_oidc/azure_service_workflow.ts

```diff
@@ -0,0 +1,92 @@
+import { MongoAzureError } from '../../../error';
+import { request } from '../../../utils';
+import type { MongoCredentials } from '../mongo_credentials';
+import { AzureTokenCache } from './azure_token_cache';
+import { ServiceWorkflow } from './service_workflow';
+
+/** Base URL for getting Azure tokens. */
+const AZURE_BASE_URL =
+  'http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01';
+
+/** Azure request headers. */
+const AZURE_HEADERS = Object.freeze({ Metadata: 'true', Accept: 'application/json' });
+
+/** Invalid endpoint result error. */
+const ENDPOINT_RESULT_ERROR =
+  'Azure endpoint did not return a value with only access_token and expires_in properties';
+
+/** Error for when the token audience is missing in the environment. */
+const TOKEN_AUDIENCE_MISSING_ERROR =
+  'TOKEN_AUDIENCE must be set in the auth mechanism properties when PROVIDER_NAME is azure.';
+
+/**
+ * The Azure access token format.
+ * @internal
+ */
+export interface AzureAccessToken {
+  access_token: string;
+  expires_in: number;
+}
+
+/**
+ * Device workflow implementation for Azure.
+ *
+ * @internal
+ */
+export class AzureServiceWorkflow extends ServiceWorkflow {
+  cache: AzureTokenCache;
```

Comment:
```suggestion
  cache = new AzureTokenCache();;
```
If you do this you can delete the `constructor`, just for succinct reasons